source/base/data_out_base.cc

```diff
@@ -7741,19 +7746,60 @@ DataOutInterface<dim, spacedim>::write_vtu_in_parallel(
                                   vtk_flags,
                                   ss);
 
-    ierr = MPI_File_write_ordered(
-      fh, ss.str().c_str(), ss.str().size(), MPI_CHAR, MPI_STATUS_IGNORE);
+    // using prefix sum to find specific offset to write at.
+    const std::uint64_t size_on_proc = ss.str().size();
+    std::uint64_t prefix_sum = 0;
+    ierr =
+      MPI_Exscan(&size_on_proc, &prefix_sum, 1, MPI_UINT64_T, MPI_SUM, comm);
     AssertThrowMPI(ierr);
+
+    // Locate specific offset for each processor.
+    const MPI_Offset offset = static_cast<MPI_Offset>(header_size) + prefix_sum;
+
+    ierr =
+      Utilities::MPI::LargeCount::MPI_File_write_at_all_c(fh,
+                                                          offset,
+                                                          ss.str().c_str(),
+                                                          ss.str().size(),
+                                                          MPI_CHAR,
+                                                          MPI_STATUS_IGNORE);
+    AssertThrowMPI(ierr);
+
+    // Sending the offset for writing the footer to rank 0.
+    if (myrank == n_ranks - 1)
+      {
+        const std::uint64_t footer_offset = size_on_proc + offset;
+        AssertThrowMPI(ierr);
+      }
```

Comment:
```suggestion
    if (myrank == n_ranks - 1)
        footer_offset = size_on_proc + offset;
```

Reply:
You never set the ``footer_offset`` correctly (you ended up creating a local variable here and setting it).

Reply:
I see. Let me changed that and test it again, thanks for pointing out!

Reply:
The test mpirun=3 still freezes, I'm wondering if there's a way to know where exactly it freezes.

Reply:
There is, of course. I will show you next time.

Reply:
sounds good! I just read about the sendrecv_replace function and it seems like I might misordered the sender and the receiver, I'm testing on that fix now.